core/src/main/java/io/grpc/internal/PickDetailsConsumerImpl.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2024 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.internal;
+
+import com.google.common.base.Preconditions;
+import io.grpc.ClientStreamTracer;
+import io.grpc.LoadBalancer.PickDetailsConsumer;
+
+/**
+ * Adapter for tracers into details consumers.
+ */
+final class PickDetailsConsumerImpl implements PickDetailsConsumer {
```

Comment:
Would a more meaningful name be `TracingPickDetailsConsumer`?

Reply:
I was torn by the name. I'm not wild about this name, but I'm also expecting we might add in other things to this interface in the future. That's why it has a default implementation, so it isn't a functional interface/lambdas don't work for it.

Reply:
That is to say, while the details today go to the tracer, other things may not.